sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -2854,10 +2854,11 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
         "dataColumns" -> query.output.map(c => toSQLId(c.name)).mkString(", ")))
   }
 
-  def tableIsNotViewError(name: TableIdentifier, replace: Boolean): Throwable = {
+  def unsupportedCreateOrReplaceViewOnTableError(
+      name: TableIdentifier, replace: Boolean): Throwable = {
     val operation = if (replace) "CREATE OR REPLACE VIEW" else "CREATE VIEW"
```

Comment:
If `replace = false`, it's better to throw `TABLE_OR_VIEW_ALREADY_EXISTS`. Otherwise, `EXPETCT_VIEW_NOT_TABLE` should be good.